File: mifosng-android/src/main/java/com/mifos/objects/response/SaveResponse.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.objects.response;


import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;
import com.mifos.api.model.Changes;

public class SaveResponse {

    @SerializedName("groupId")
    Integer groupId;

    @SerializedName("resourceId")
    Integer resourceId;

    @SerializedName("officeId")
    Integer officeId;

    Changes changes;

    @Override
    public String toString() {
        return new Gson().toJson(this);
    }

    public Integer getGroupId() {
        return groupId;
    }

    public void setGroupId(Integer groupId) {
        this.groupId = groupId;
    }

    public Integer getResourceId() {
        return resourceId;
    }

    public void setResourceId(Integer resourceId) {
        this.resourceId = resourceId;
    }

    public Integer getOfficeId() {
        return officeId;
    }

    public void setOfficeId(Integer officeId) {
        this.officeId = officeId;
    }

    public Changes getChanges() {
        return changes;
    }

    public void setChanges(Changes changes) {
        this.changes = changes;
    }
}

File: core/model/src/commonMain/kotlin/com/mifos/core/model/objects/responses/SaveResponse.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.model.objects.responses

import com.mifos.core.model.objects.Changes
import com.mifos.core.model.utils.IgnoredOnParcel
import com.mifos.core.model.utils.Parcelable
import com.mifos.core.model.utils.Parcelize
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json

@Serializable
@Parcelize
class SaveResponse(
    var groupId: Int? = null,

    var resourceId: Int? = null,

    var officeId: Int? = null,

    @IgnoredOnParcel
    var changes: Changes? = null,
) : Parcelable {
    override fun toString(): String {
        return Json.encodeToString(serializer(), this)
    }
}
